rust/agama-lib/src/storage/client/zfcp.rs

```diff
@@ -0,0 +1,207 @@
+    }
+
+    pub async fn is_lun_scan_allowed(&self) -> Result<bool, ServiceError> {
+        let allowed = self.manager_proxy.allow_lunscan().await?;
+        // simply check if introspection contain given interface
+        Ok(allowed)
+    }
+
+    pub async fn probe(&self) -> Result<(), ServiceError> {
+        Ok(self.manager_proxy.probe().await?)
+    }
+
+    pub async fn get_disks(&self) -> Result<Vec<(OwnedObjectPath, ZFCPDisk)>, ServiceError> {
+        let managed_objects = self.object_manager_proxy.get_managed_objects().await?;
+
+        let mut devices: Vec<(OwnedObjectPath, ZFCPDisk)> = vec![];
+        for (path, ifaces) in managed_objects {
+            if let Some(properties) = ifaces.get("org.opensuse.Agama.Storage1.ZFCP.Disk") {
+                match ZFCPDisk::try_from(properties) {
+                    Ok(device) => {
+                        devices.push((path, device));
+                    }
+                    Err(error) => {
+                        log::warn!("Not a valid zFCP disk: {}", error);
+                    }
+                }
+            }
+        }
+        Ok(devices)
+    }
+
+    pub async fn get_controllers(
+        &self,
+    ) -> Result<Vec<(OwnedObjectPath, ZFCPController)>, ServiceError> {
+        let managed_objects = self.object_manager_proxy.get_managed_objects().await?;
+
+        let mut devices: Vec<(OwnedObjectPath, ZFCPController)> = vec![];
+        for (path, ifaces) in managed_objects {
+            if let Some(properties) = ifaces.get("org.opensuse.Agama.Storage1.ZFCP.Controller") {
+                // TODO: maybe move it to model? but it needs also additional calls to dbus
+                let path_s = Self::controller_id_from_path(&path);
+                devices.push((
+                    path,
+                    ZFCPController {
+                        id: path_s.clone(),
+                        channel: get_property(properties, "Channel")?,
+                        lun_scan: get_property(properties, "LUNScan")?,
+                        active: get_property(properties, "Active")?,
+                        luns_map: self.get_luns_map(path_s.as_str()).await?,
+                    },
+                ))
+            }
+        }
+        Ok(devices)
+    }
+
+    /// Gets controller id that can be used in other controller methods from its dbus path
+    ///
+    /// common case:
+    /// ```
+    ///     let path = zbus::zvariant::OwnedObjectPath::try_from("/test/controllers/2").unwrap();
+    ///     let id = agama_lib::storage::client::zfcp::ZFCPClient::controller_id_from_path(&path);
+    ///     assert_eq!(id, "2".to_string());
+    /// ```
+    ///
+    /// edge case with malformed path that will panic in Owned path already
+    /// ```should_panic
+    ///     let path = zbus::zvariant::OwnedObjectPath::try_from("mangled").unwrap();
+    ///     let id = agama_lib::storage::client::zfcp::ZFCPClient::controller_id_from_path(&path);
```

Comment:
According to this comment, the call to `OwnedPath::try_from` will fail, right? So what is the point of this code?